packages/@react-aria/overlays/src/ariaHideOutside.ts

```diff
@@ -26,6 +33,62 @@ let observerStack = [];
 export function ariaHideOutside(targets: Element[], root = document.body) {
   let visibleNodes = new Set<Element>(targets);
   let hiddenNodes = new Set<Element>();
+  const shadowRoots = findShadowRoots(targets);
+
+  if (shadowRoots.length > 0) {
+    const targetsByShadowRoot = new Map<ShadowRoot, Element[]>();
```

Comment:
The function `ariaHideOutside` already accepts a `root` element. Can we make this generic so that any element can be treated as root and not just the `document.body` or shadow root ? If an element passed as `root` is inside shadow root, even that should work. Does this make sense ?

This means that all the hooks using this function should also accept a custom root and should pass this information to this element.